# Importing the Libraries


In [2]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns

from sklearn.experimental import enable_iterative_imputer
from sklearn.impute import IterativeImputer
from sklearn.impute import SimpleImputer
from sklearn.feature_extraction.text import CountVectorizer
from scipy import stats

from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import classification_report

C:\Users\ashut\AppData\Local\Programs\Python\Python39\lib\site-packages\scipy\__init__.py:177: UserWarning: A NumPy version >=1.18.5 and <1.26.0 is required for this version of SciPy (detected version 1.26.0
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"


# Datasets


In [3]:
# Importing the Philadelphia buildings dataset

df1 = pd.read_csv('phl_buildings.csv')
df1.head()

OBJECTID      BIN  FCODE         ADDRESS BUILDING_NAME  BASE_ELEVATION  \
0  51065281  1000001   1810  1501 N 18TH ST           NaN           104.2   
1  51065282  1000002   1810  1501 N 18TH ST           NaN           103.3   
2  51065283  1000003   1810  1501 N 18TH ST           NaN           103.9   
3  51065284  1000004   1810  1501 N 18TH ST           NaN           103.6   
4  51065285  1000005   1810  1501 N 18TH ST           NaN           108.0   

   APPROX_HGT  MAX_HGT PARCEL_ID_NUM PARCEL_ID_SOURCE  Shape__Area  \
0        20.0     28.0        319860              PWD    97.761719   
1        20.0     29.0        319860              PWD    99.632812   
2        20.0     28.0        319860              PWD    96.292969   
3        20.0     28.0        319860              PWD    98.699219   
4        20.0     30.0        319860              PWD   101.902344   

   Shape__Length  
0      40.383631  
1      40.691884  
2      40.145485  
3      40.543015  
4      42.790290

In [4]:
print(f"No. of rows in df1:", len(df1))
print(f"No. columns in df1:", len(df1.columns))

No. of rows in df1: 543278
No. columns in df1: 12


In [5]:
# Importing the Philadelphia Properties dataset
df2 = pd.read_csv("phl_properties.csv", low_memory = False)

print(f"No. of rows in df2:", len(df2))
print(f"No. columns in df2:", len(df2.columns))

No. of rows in df2: 581456
No. columns in df2: 75


<b> Since we are interested in in property prie prediction. That's why we are nly going to use the Philadelphia Properties dataset. </b>


In [6]:
df = df2.copy()

In [7]:
# Removing the columns which do not contribute to the property price

df.drop(columns=['assessment_date','beginning_point','book_and_page','census_tract','cross_reference','date_exterior_condition',
                 'exempt_building','exempt_land','geographic_ward','homestead_exemption','house_extension','house_number',
                 'mailing_zip','market_value_date','off_street_open','separate_utilities','sewer','site_type','state_code',
                 'street_code','street_direction','street_name','suffix','taxable_building','taxable_land','utility',
                 'year_built_estimate','objectid','mailing_address_1','mailing_address_2','mailing_care_of','mailing_city_state',
                 ], inplace=True)

In [8]:
df.head()

basements building_code building_code_description  category_code  \
0         D         P50     ROW W/GAR 3 STY MASONRY              1   
1         D         O50           ROW 3 STY MASONRY              1   
2         D         O50           ROW 3 STY MASONRY              1   
3         D         O50           ROW 3 STY MASONRY              1   
4         C         O50           ROW 3 STY MASONRY              1   

  category_code_description central_air  depth  exterior_condition  \
0             Single Family           N   49.0                 5.0   
1             Single Family           Y   49.0                 2.0   
2             Single Family           N   49.0                 4.0   
3             Single Family           Y  100.0                 4.0   
4             Single Family           N  100.0                 4.0   

   fireplaces  frontage  ... topography  total_area total_livable_area  \
0         0.0     15.90  ...          F       779.0             1650.0   
1         0.0     15.90  ...          F       779.1             1203.0   
2         0.0     14.80  ...          F       725.2             1482.0   
3         0.0     14.33  ...          F      1433.0             2049.0   
4         0.0     15.00  ...          F      1500.0             1155.0   

  type_heater  unfinished unit view_type  year_built   zip_code  zoning  
0           H         NaN  NaN         I        1920  191475425   RSA5   
1           H         NaN  NaN         I        1920  191475425   RSA5   
2           H         NaN  NaN         I        1920  191475425   RSA5   
3           A         NaN  NaN         I        1920  191475425   RSA5   
4           H         NaN  NaN         I        1920  191475425   RSA5   

[5 rows x 43 columns]

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 581456 entries, 0 to 581455
Data columns (total 43 columns):
 #   Column                     Non-Null Count   Dtype  
---  ------                     --------------   -----  
 0   basements                  325571 non-null  object 
 1   building_code              581456 non-null  object 
 2   building_code_description  581456 non-null  object 
 3   category_code              581456 non-null  int64  
 4   category_code_description  581456 non-null  object 
 5   central_air                285860 non-null  object 
 6   depth                      581454 non-null  float64
 7   exterior_condition         556191 non-null  float64
 8   fireplaces                 581435 non-null  float64
 9   frontage                   581454 non-null  float64
 10  fuel                       10900 non-null   object 
 11  garage_spaces              581435 non-null  float64
 12  garage_type                530522 non-null  object 
 13  general_construction       51

To ease the exploratory data analysis process, we grouped the columns based on its types and its connections with other columns.

1. Property Location:
    * Location, Street Designation
2. Classification of Property:
    * Building code, Building code description, Category code, Category code description
    * Unit
    * Zoning
    * Unfinished
    * Zip code
3. Property Specifications:
    * Basements, Garage type, Garage spaces
    * Central air, Fireplaces, Other building, Unfinished, Topography, View type
    * Depth, Frontage, Total area, Total livable area
    * Exterior condition, Interior condition, Quality grade, General construction
    * Fuel, Type heater
    * Number of bathrooms, Number of bedrooms, Number of rooms, Number stories, Year built 
    * Market value, Sale price, Sale date
4. Others:
    * Parcel number, Parcel shape
    * Recording date, registry number


# Cleaning and Initial EDA of Features

<strong> Category Code, Building Code and Zoning </strong>

In [10]:
df[['category_code','category_code_description']].dtypes

category_code                 int64
category_code_description    object
dtype: object

In [11]:
df['category_code'].value_counts()

1    461573
6     44612
2     42462
3     14396
4     14059
5      4354
Name: category_code, dtype: int64

In [12]:
df['category_code_description'].value_counts()

Single Family    461573
Vacant Land       44612
Multi Family      42462
Mixed Use         14396
Commercial        14059
Industrial         4354
Name: category_code_description, dtype: int64

<b> Building Code </b>

In [13]:
df[['building_code','building_code_description']].dtypes

building_code                object
building_code_description    object
dtype: object

In [14]:
df[['building_code','building_code_description']].nunique()

building_code                802
building_code_description    798
dtype: int64

<b> Zoning </b>


In [15]:
df['zoning'].dtypes

dtype('O')

In [16]:
df['zoning'].nunique()

43

In [17]:
df['zoning'] = df['zoning'].apply(lambda x: str(x).strip())
df['zoning'].unique()

array(['RSA5', 'ICMX', 'RM1', 'CMX1', 'I2', 'CMX2', 'RMX2', 'CMX25',
       'CA1', 'nan', 'CMX3', 'SPPOA', 'RM2', 'RSA3', 'CMX2.', 'CMX4',
       'RSA1', 'RSD1', 'IRMX', 'RMX3', 'CMX5', 'RM4', 'I1', 'RS3', 'RSA2',
       'RTA1', 'RSD3', 'RMX1', 'RM3', 'RSA', 'SPINS', 'RSD2', 'CA2', 'I3',
       'SPAIR', 'RSA4', '12', 'IP', 'SPSTA', 'SC', 'SPENT', '2002',
       'SPPOP', 'SP'], dtype=object)

In [18]:
df['zoning'].value_counts()

RSA5     288769
RM1      105876
RSA3      62750
CMX2      20772
RSD3      13180
RSA2      12602
RSA4      12188
CMX3       8053
RTA1       5766
RM2        5547
CMX1       5445
CMX5       4915
CMX25      4335
I2         4270
CMX4       4194
RM4        3814
RSD1       3270
ICMX       2993
RMX3       2276
RSA1       1294
RM3        1292
RMX1       1275
RSD2       1274
SPPOA       868
CA1         844
I1          725
IRMX        708
CA2         628
I3          433
SPINS       366
RMX2        271
CMX2.       211
nan         133
12           45
SPAIR        23
IP           23
SPSTA        16
RSA           3
SPENT         3
SPPOP         2
2002          1
RS3           1
SC            1
SP            1
Name: zoning, dtype: int64

<b> Dropping Anomalies </b>


In [19]:
index = df[df['zoning']=='2002'].index
df.drop(index=index, axis=0, inplace=True)

In [20]:
index = df[df['zoning']=='SP'].index
df.drop(index=index, axis=0, inplace=True)

In [21]:
index = df[df['zoning']=='RS3'].index
df.drop(index=index, axis=0, inplace=True)

In [22]:
index = df[df['zoning']=='SC'].index
df.drop(index=index, axis=0, inplace=True)

In [23]:
index = df[df['zoning']=='RSA'].index
df.drop(index=index, axis=0, inplace=True)

<b> Fixing Typos </b>

In [24]:
df['zoning'] = df['zoning'].apply(lambda x: 'I2' if x=='12' else x)

In [25]:
df['zoning'] = df['zoning'].apply(lambda x: 'CMX2' if x=='CMX2' else x)


In [26]:
df['zoning'] = df['zoning'].apply(lambda x: np.nan if x=='nan' else x)


In [27]:
df['zoning'].isna().sum()

133

<b> Dropping NAs </b>

In [28]:
index = df[df['zoning'].isna()].index
df.drop(index=index, axis=0, inplace=True)

In [29]:
df['zoning'].isna().sum()

0

In [30]:
df['new_zoning']=df['zoning'].apply(lambda x: 'Residential Single Family' if (x=='RSD1') | (x=='RSD2') | (x=='RSD3') 
                   | (x=='RSA1') | (x=='RSA2') | (x=='RSA3') | (x=='RSA4') | (x=='RSA5') else 
                   'Residential Two Family' if (x=='RTA1') else
                   'Residential Multi Family' if (x=='RM1') | (x=='RM2') | (x=='RM3') | (x=='RM4') else
                   'Mixed Used' if (x=='RMX1') | (x=='RMX2') | (x=='RMX3') | (x=='CMX1') | (x=='CMX2') 
                   | (x=='CMX25') | (x=='CMX3') | (x=='CMX4') | (x=='CMX5') | (x=='ICMX') | (x=='IRMX')else
                   'Commercial' if (x=='CA1') | (x=='CA2') else
                   'Industrial' if (x=='I1') | (x=='I2') | (x=='I3') | (x=='IP') else 
                   'Special Purpose Airport' if (x=='SPAIR') else 
                   'Special Purpose Institution' if (x=='SPINS') else 
                   'Special Purpose Entertainment' if (x=='SPENT') else 
                   'Special Purpose Stadium' if (x=='SPSTA') else 
                   'Special Purpose Open Space-Active' if (x=='SPPOA') else 
                   'Special Purpose Open Space-Passive' if (x=='SPPOP') else '0')

In [31]:
df['new_zoning'].value_counts()

Residential Single Family             395327
Residential Multi Family              116529
Mixed Used                             55237
Residential Two Family                  5766
Industrial                              5496
Commercial                              1472
Special Purpose Open Space-Active        868
Special Purpose Institution              366
0                                        211
Special Purpose Airport                   23
Special Purpose Stadium                   16
Special Purpose Entertainment              3
Special Purpose Open Space-Passive         2
Name: new_zoning, dtype: int64

<strong> Crosschecking Category and Building Code </strong>

In [32]:
df['new_zoning']

0         Residential Single Family
1         Residential Single Family
2         Residential Single Family
3         Residential Single Family
4         Residential Single Family
                    ...            
581451                   Mixed Used
581452                   Mixed Used
581453                   Mixed Used
581454                   Mixed Used
581455                   Mixed Used
Name: new_zoning, Length: 581316, dtype: object

In [33]:
pd.set_option('display.max_rows',802)
df.groupby('category_code_description')['building_code_description'].value_counts()

category_code_description  building_code_description 
Commercial                 STORE 1 STY MASONRY             1549
                           AUTO REPAIR SHOP MASONRY        1109
                           HSE WORSHIP ALL 2 STY MAS        740
                           COM CONDO 1 STY MASONRY          721
                           HSE WORSHIP ALL 1 STY MAS        552
                           STORE 2 STY MASONRY              462
                           STR/OFF 2 STY MASONRY            455
                           STR/OFF 1 STY MASONRY            434
                           MISC DAY CARE MASONRY            353
                           HSE WORSHIP ALL 3 STY MAS        327
                           OFF BLD N/COM W/PKG MASON        294
                           OFF BLD N/PKG N/COM MASON        273
                           SCHOOL 3 STY MASONRY             271
                           REST'RNT W/BAR MASONRY           261
                           STORE 3 STY MASONRY    

In [34]:
def checking(c):
    c = c.str.extractall('([a-zA-Z]+)')
    c.columns=['check']
    b = c.reset_index(drop=True)
    c = b['check'].value_counts()
    
    checking = {'checking':c.index, 'freq':c.values}
    checking = pd.DataFrame(checking)
    checking.index = checking['checking']
    checking.drop('checking',axis=1,inplace=True)
    checking.sort_values('freq',ascending=False,inplace=True)
    return checking

In [35]:
check_1=checking(df[df['category_code']==1]['building_code_description'])
check_2=checking(df[df['category_code']==2]['building_code_description'])
check_3=checking(df[df['category_code']==3]['building_code_description'])
check_4=checking(df[df['category_code']==4]['building_code_description'])
check_5=checking(df[df['category_code']==5]['building_code_description'])
check_6=checking(df[df['category_code']==6]['building_code_description'])

In [36]:
check_1.rename(columns={'freq':'Single Family'},inplace=True)
check_2.rename(columns={'freq':'Multi Family'},inplace=True)
check_3.rename(columns={'freq':'Mixed Use'},inplace=True)
check_4.rename(columns={'freq':'Commercial'},inplace=True)
check_5.rename(columns={'freq':'Industrial'},inplace=True)
check_6.rename(columns={'freq':'Vacant Land'},inplace=True)

In [37]:
check_count = pd.concat([check_1,check_2,check_3,check_4,check_5,check_6],axis=1)
check_count.fillna(0,inplace=True)

In [38]:
check_count

Single Family  Multi Family  Mixed Use  Commercial  Industrial  \
checking                                                                       
STY                459237.0       37284.0    14396.0      7336.0         4.0   
MASONRY            406197.0        5151.0     6867.0      9618.0      3966.0   
ROW                332555.0       16173.0     7051.0         0.0         0.0   
GAR                177091.0           0.0        0.0       369.0         0.0   
B                  145700.0           0.0      797.0         0.0         0.0   
W                   66418.0           0.0     6303.0       814.0         0.0   
DET                 56920.0        1008.0      218.0         0.0         0.0   
D                   51978.0        5318.0      440.0         0.0         0.0   
S                   38755.0        5318.0      440.0         5.0         0.0   
CONDO               34656.0           0.0        0.0       932.0       101.0   
RES                 32347.0           0.0        0.0         0.0         0.0   
MAS                 28329.0        2800.0      197.0      2531.0        65.0   
SEMI                26877.0           0.0        0.0         0.0         0.0   
OTH                 15138.0           5.0       46.0       434.0       558.0   
FRAME               13493.0         289.0       33.0        40.0        10.0   
OTHER               12211.0          61.0        0.0       173.0       146.0   
STONE                8800.0         597.0      140.0         0.0         0.0   
PARKING              2273.0           0.0        0.0         0.0         0.0   
SPACE                2273.0           0.0        0.0         0.0         0.0   
MASON                2001.0       33314.0     7154.0      1044.0       146.0   
PRIV                 1507.0           0.0        0.0       194.0         0.0   
O                    1171.0        1716.0       54.0       281.0        28.0   
OTHE                  463.0           0.0        0.0         3.0         0.0   
LOFT                  385.0           0.0        0.0         0.0        53.0   
STACKED                52.0           0.0        0.0         0.0         0.0   
PUD                    52.0           0.0        0.0         0.0         0.0   
STORAGE                36.0           0.0        0.0         0.0        35.0   
UNIT                   36.0           0.0        0.0         0.0         0.0   
STO                    30.0           3.0        1.0         0.0         0.0   
AIR                     8.0           0.0        0.0        55.0         0.0   
RIGHTS                  8.0           0.0        0.0        41.0         0.0   
RESIDENTIAL             8.0           0.0        0.0         0.0         0.0   
DWELL                   5.0           0.0        0.0         0.0         0.0   
MISC                    5.0           0.0        0.0      1059.0         0.0   
FRM                     2.0          10.0        0.0        18.0         0.0   
APT                     0.0       37260.0     6687.0         0.0         0.0   
CONV                    0.0       22499.0        0.0         0.0         0.0   
UNITS                   0.0       19434.0        0.0         0.0         0.0   
APTS                    0.0        4972.0        0.0         0.0         0.0   
OT                      0.0         328.0        0.0        10.0         0.0   
HOTEL                   0.0         164.0        0.0         0.0         0.0   
STON                    0.0         154.0        0.0         0.0         0.0   
DORMITORY               0.0         152.0        0.0         0.0         0.0   
HOME                    0.0         147.0        0.0       205.0         0.0   
BOARDING                0.0         147.0        0.0         0.0         0.0   
RM                      0.0          75.0        0.0         0.0         0.0   
HSE                     0.0          75.0        0.0      1780.0         0.0   
FRAM                    0.0          62.0        0.0         1.0         0.0   
MA            

By skimming two tables above, it seems that the category code was classified using building code description. But, since the data is a lot to check with, we will answer the question using a simple modeling.

In [39]:
text = df['building_code_description']
label = df['category_code_description']


In [40]:
text_train, text_test, y_train, y_test = train_test_split(text, label, stratify=label, random_state=2020)

In [41]:
vect = CountVectorizer().fit(text_train)

In [42]:
x_train = vect.transform(text_train)
x_test = vect.transform(text_test)

In [43]:
feature = vect.get_feature_names_out()

In [44]:
model = LogisticRegression(n_jobs=-1)
model.fit(x_train, y_train)
y_pred = model.predict(x_test)

In [45]:
print(classification_report(y_test, y_pred))

               precision    recall  f1-score   support

   Commercial       0.98      1.00      0.99      3513
   Industrial       1.00      1.00      1.00      1088
    Mixed Use       1.00      0.98      0.99      3599
 Multi Family       1.00      1.00      1.00     10601
Single Family       1.00      1.00      1.00    115388
  Vacant Land       1.00      1.00      1.00     11140

     accuracy                           1.00    145329
    macro avg       1.00      1.00      1.00    145329
 weighted avg       1.00      1.00      1.00    145329



This model proves that building code description can specifically predict which category code the data belongs to. It's very specific that it proves that there are no misclassification of category code in the dataset.

This also confirms that category code is representing the type of the property, not the building. For example, logically, condominium (CONDO) should be classified into multi family if category code represents the building.

In [46]:
df[df['building_code_description'].str.contains('CONDO')].value_counts('category_code_description')

category_code_description
Single Family    34656
Commercial         932
Industrial         101
dtype: int64

Changing the Category Code Description to following.
* Resident
* Hotel and Apartments
* Store and Dwelling
*  Commercial
*  Industrial
*  Vacant Land

In [48]:
dupe = pd.read_csv('./Data/dupe.csv')
dupe.drop(columns=['Unnamed: 0'], inplace=True)
len(dupe)

3634

In [49]:
df_dupe = df[df['location'].isin(dupe['dupe'])]
len(df_dupe)

37692

In [50]:
len(df_dupe)/len(df)*100

6.483908923889932

We have 37692 (6.5% from total) data with duplicate 'location'.

# Crosschecking

In [52]:
df['unit'].dtypes

dtype('O')

In [54]:
df['unit'].nunique()

8761

In [55]:
len(df[df['unit'].notna()])

38717

In [56]:
len(df_dupe[df_dupe['unit'].notna()])/len(df[df['unit'].notna()])*100

96.19546969031691

Most of the data with filled 'unit' value were located in data with duplicate 'location' value (df_dupe).

In [57]:
check = df_dupe[['location','unit','sale_date']].sort_values(by=['location','unit'])
check[0:100]

location     unit            sale_date
575439  1 ACADEMY CIR  00000P1  2009-08-03 00:00:00
570503  1 ACADEMY CIR  00000P2  2011-08-12 00:00:00
575440  1 ACADEMY CIR  00000P3  2016-04-22 00:00:00
575441  1 ACADEMY CIR  00000P4  2016-07-01 00:00:00
575442  1 ACADEMY CIR  00000P5  2007-01-25 00:00:00
575443  1 ACADEMY CIR  00000P6  2006-11-17 00:00:00
575444  1 ACADEMY CIR  00000P7  2009-09-29 00:00:00
575445  1 ACADEMY CIR  00000P8  2006-10-23 00:00:00
575446  1 ACADEMY CIR  00000P9  2006-10-25 00:00:00
573269  1 ACADEMY CIR  0000101  2019-06-17 00:00:00
573270  1 ACADEMY CIR  0000102  2006-09-07 00:00:00
573271  1 ACADEMY CIR  0000103  2015-12-03 00:00:00
573272  1 ACADEMY CIR  0000104  2008-05-12 00:00:00
573273  1 ACADEMY CIR  0000105  2017-09-18 00:00:00
573274  1 ACADEMY CIR  0000106  2016-11-10 00:00:00
573479  1 ACADEMY CIR  0000107  2008-05-12 00:00:00
573480  1 ACADEMY CIR  0000108  2007-05-16 00:00:00
573481  1 ACADEMY CIR  0000109  2012-06-14 00:00:00
573482  1 ACADEMY CIR  0000110  2017-12-29 00:00:00
573483  1 ACADEMY CIR  0000111  2017-11-18 00:00:00
573484  1 ACADEMY CIR  0000112  2016-12-01 00:00:00
573485  1 ACADEMY CIR  0000113  2017-01-13 00:00:00
573486  1 ACADEMY CIR  0000114  2008-05-12 00:00:00
573487  1 ACADEMY CIR  0000115  2013-06-12 00:00:00
573488  1 ACADEMY CIR  0000116  2013-02-15 00:00:00
573489  1 ACADEMY CIR  0000117  2006-10-24 00:00:00
573490  1 ACADEMY CIR  0000118  2008-05-12 00:00:00
573491  1 ACADEMY CIR  0000119  2006-09-29 00:00:00
573696  1 ACADEMY CIR  0000120  2015-11-02 00:00:00
573697  1 ACADEMY CIR  0000201  2006-09-27 00:00:00
573698  1 ACADEMY CIR  0000202  2012-08-03 00:00:00
573699  1 ACADEMY CIR  0000203  2008-01-17 00:00:00
573700  1 ACADEMY CIR  0000204  2008-05-12 00:00:00
573701  1 ACADEMY CIR  0000205  2020-02-26 00:00:00
573702  1 ACADEMY CIR  0000206  2018-10-26 00:00:00
573703  1 ACADEMY CIR  0000207  2008-05-12 00:00:00
573704  1 ACADEMY CIR  0000208  2007-08-27 00:00:00
573705  1 ACADEMY CIR  0000209  2007-07-10 00:00:00
573706  1 ACADEMY CIR  0000210  2006-10-25 00:00:00
573707  1 ACADEMY CIR  0000211  2006-10-23 00:00:00
573708  1 ACADEMY CIR  0000212  2015-09-30 00:00:00
573910  1 ACADEMY CIR  0000213  2006-11-17 00:00:00
573911  1 ACADEMY CIR  0000214  2019-10-07 00:00:00
573912  1 ACADEMY CIR  0000215  2017-05-15 00:00:00
573913  1 ACADEMY CIR  0000216  2019-11-18 00:00:00
573914  1 ACADEMY CIR  0000217  2006-10-27 00:00:00
573915  1 ACADEMY CIR  0000218  2006-11-20 00:00:00
573916  1 ACADEMY CIR  0000219  2013-12-23 00:00:00
573917  1 ACADEMY CIR  0000220  2018-07-25 00:00:00
573918  1 ACADEMY CIR  0000221  2011-11-12 00:00:00
573919  1 ACADEMY CIR  0000222  2011-09-30 00:00:00
573920  1 ACADEMY CIR  0000301  2017-04-28 00:00:00
573921  1 ACADEMY CIR  0000302  2014-03-21 00:00:00
573922  1 ACADEMY CIR  0000303  2015-08-18 00:00:00
573923  1 ACADEMY CIR  0000304  2014-12-05 00:00:00
574127  1 ACADEMY CIR  0000305  2018-09-28 00:00:00
574128  1 ACADEMY CIR  0000306  2008-05-12 00:00:00
574129  1 ACADEMY CIR  0000307  2019-08-02 00:00:00
574130  1 ACADEMY CIR  0000308  2011-07-22 00:00:00
574131  1 ACADEMY CIR  0000309  2019-06-26 00:00:00
574132  1 ACADEMY CIR  0000310  2016-06-29 00:00:00
574133  1 ACADEMY CIR  0000311  2019-07-18 00:00:00
574134  1 ACADEMY CIR  0000312  2010-08-13 00:00:00
574135  1 ACADEMY CIR  0000313  2020-01-02 00:00:00
574136  1 ACADEMY CIR  0000314  2016-04-22 00:00:00
574137  1 ACADEMY CIR  0000315  2018-07-13 00:00:00
574138  1 ACADEMY CIR  0000316  2019-01-02 00:00:00
574139  1 ACADEMY CIR  0000317  2018-03-12 00:00:00
574140  1 ACADEMY CIR  0000318  2016-10-03 00:00:00
574342  1 ACADEMY CIR  0000319  2020-01-15 00:00:00
574343  1 ACADEMY CIR  0000320  2016-06-11 00:00:00
574344  1 ACADEMY CIR  0000321  2019-05-23 00:00:00
574345  1 ACADEMY CIR  0000322  2011-08-19 00:00:00
574346  1 ACADEMY CIR  0000323  2008-05-12 00:00:00
574347  1 ACADEMY CIR  0000324  2010-05-13 00:00:00
575447  1 ACADEMY CIR  0000P10  2013-05-06 00:00:00
570286  1

In [59]:
location = []
for i in check['location'].unique():
    unit = []
    for j in check[check['location']==i]['unit']:
        if j not in unit:
            unit.append(j)
        elif j in unit:
            location.append(i)
location

['1100 S CHRIS COLUMBUS BLV',
 '1126-36 ARCH ST',
 '115 SOUTH ST',
 '1341 S CHRIS COLUMBUS BLV',
 '1401 S CHRIS COLUMBUS BLV',
 '1500 JOHN F KENNEDY BLVD',
 '1801 JOHN F KENNEDY BLVD',
 '1801 W HUNTING PARK AVE',
 '2906 S CHRIS COLUMBUS BLV',
 '3701 S CHRIS COLUMBUS BLV',
 '4101 S CHRIS COLUMBUS BLV',
 '4201 WALNUT ST',
 '700 WALNUT LN',
 '7901-03 FRANKFORD AVE',
 '8998 PINE RD',
 '948 N LAWRENCE ST']